cmd/helper/main.go

```diff
@@ -18,30 +18,80 @@ package main
 
 import (
 	"fmt"
+	"net/http"
+	"net/url"
 	"os"
 
 	"github.com/buildpacks/libcnb"
+
 	"github.com/paketo-buildpacks/libpak/bard"
+	"github.com/paketo-buildpacks/libpak/bindings"
+	"github.com/paketo-buildpacks/microsoft-azure/appinsights"
+	"github.com/paketo-buildpacks/microsoft-azure/credentials"
+	"github.com/paketo-buildpacks/microsoft-azure/internal/common"
 
 	"github.com/paketo-buildpacks/libpak/sherpa"
-
-	"github.com/paketo-buildpacks/azure-application-insights/helper"
 )
 
 func main() {
 	sherpa.Execute(func() error {
-		var (
-			err error
-			p   = helper.Properties{Logger: bard.NewLogger(os.Stdout)}
-		)
+		l := bard.NewLogger(os.Stdout)
 
-		p.Bindings, err = libcnb.NewBindingsFromEnvironment()
+		a, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("unable to read working directory\n%w", err)
+		}
+
+		bs, err := libcnb.NewBindingsFromEnvironment()
 		if err != nil {
 			return fmt.Errorf("unable to read bindings from environment\n%w", err)
 		}
 
+		var (
+			b libcnb.Binding
+			c common.CredentialSource
+		)
+
+		if hasMetadataServer() {
+			c = common.MetadataServer
+		} else if g, ok, err := bindings.ResolveOne(bs, bindings.OfType("MicrosoftAzure")); err != nil {
+			return fmt.Errorf("unable to resolve MicrosoftAzure binding\n%w", err)
+		} else if ok {
+			b = g
```

Comment:
```suggestion
		} else if m, ok, err := bindings.ResolveOne(bs, bindings.OfType("MicrosoftAzure")); err != nil {
			return fmt.Errorf("unable to resolve MicrosoftAzure binding\n%w", err)
		} else if ok {
			b = m
```
assume the `g` is a copy pasted shorthand for google?